src/components/v2/devtronStackManager/DevtronStackManager.component.tsx

```diff
@@ -0,0 +1,548 @@
+                                </>
+                            )}
+                        {!isActionTriggered &&
+                            (installationStatus === ModuleStatus.INSTALL_FAILED ||
+                                installationStatus === ModuleStatus.UPGRADE_FAILED ||
+                                installationStatus === ModuleStatus.TIMEOUT ||
+                                installationStatus === ModuleStatus.UNKNOWN) && (
+                                <>
+                                    <RetyrInstallIcon className="module-details__retry-install-icon icon-dim-16 mr-8" />
+                                    {`Retry ${isUpgradeView ? 'update' : 'install'}`}
+                                </>
+                            )}
+                    </button>
+                )}
+            {((installationStatus !== ModuleStatus.NOT_INSTALLED && installationStatus !== ModuleStatus.HEALTHY) ||
+                (installationStatus === ModuleStatus.HEALTHY && !latestVersionAvailable)) && (
+                <InstallationStatus
+                    installationStatus={installationStatus}
+                    appName={serverInfo?.releaseName}
+                    isCICDInstalled={isCICDInstalled}
+                    logPodName={logPodName}
+                    isUpgradeView={isUpgradeView}
+                    latestVersionAvailable={latestVersionAvailable}
+                />
+            )}
+            {!isUpgradeView && installationStatus === ModuleStatus.INSTALLED && <ModuleUpdateNote />}
+            {(installationStatus === ModuleStatus.INSTALL_FAILED ||
+                installationStatus === ModuleStatus.UPGRADE_FAILED ||
+                installationStatus === ModuleStatus.TIMEOUT ||
+                installationStatus === ModuleStatus.UNKNOWN) && <GetHelpCard />}
+        </div>
+    )
+}
+
+export const ModuleDetailsView = ({
+    moduleDetails,
+    setDetailsMode,
+    setShowManagedByDialog,
+    serverInfo,
+    upgradeVersion,
+    logPodName,
+    fromDiscoverModules,
+    isActionTriggered,
+    handleActionTrigger,
+    history,
+    location,
+}: ModuleDetailsViewType): JSX.Element | null => {
+    useEffect(() => {
+        if (!moduleDetails && !new URLSearchParams(location.search).get('id')) {
+            setDetailsMode('')
+            history.push(
+                fromDiscoverModules ? URLS.STACK_MANAGER_DISCOVER_MODULES : URLS.STACK_MANAGER_INSTALLED_MODULES,
+            )
+        }
+    }, [])
+
+    return moduleDetails ? (
+        <div className="module-details__view-container">
+            <Carousel className="module-details__carousel mb-24" imageUrls={moduleDetails.assets} />
+            <div className="module-details__view-wrapper">
+                <div className="module-details__feature-wrapper">
+                    <h2 className="module-details__feature-heading cn-9 fs-20 fw-6">{moduleDetails?.title}</h2>
```

Comment:
We can remove '?' from moduleDetails as we are explicitly checking for moduleDetails

Reply:
Done